nntrainer/optimizers/optimizer_devel.h

```diff
@@ -125,6 +125,14 @@ class Optimizer : public ml::train::Optimizer {
    */
   virtual void checkValidation() const;
 
+  /**
+   * @brief     Add extra variables per weight if the optimizer needs any.
+   * @param weight_spec weight_spec
+   * @retval #ML_ERROR_NONE Successful.
+   * @retval #ML_ERROR_INVALID_PARAMETER invalid parameter.
+   */
+  virtual void addOptimizerVariable(Weight::Spec &weight_spec) = 0;
```

Comment:
please add a TODO that L142 interface must be removed.
same for optimizer_impl.

Reply:
Added